javalibintf/src/main/java/org/scalajs/javalibintf/TypedArrayBuffer.java

```diff
@@ -0,0 +1,370 @@
+/*
+ * Scala.js (https://www.scala-js.org/)
+ *
+ * Copyright EPFL.
+ *
+ * Licensed under Apache License 2.0
+ * (https://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package org.scalajs.javalibintf;
+
+import java.nio.*;
+
+/**
+ * Utilities to interface {@link java.nio.Buffer}s and JavaScript TypedArrays.
+ *
+ * <p>{@link java.nio.Buffer}s can be <em>direct</em> buffers or
+ * <em>indirect</em> buffers. Indirect buffers use an underlying array (like
+ * {@code int[]} in Java or {@code Array[Int]} in Scala). Direct buffers are
+ * supposed to use off-heap memory.
+ *
+ * <p>In a JavaScript environment, the equivalent of off-heap memory for
+ * buffers of primitive numeric types are TypedArrays.
+ *
+ * <p>This class provides methods to wrap TypedArrays as direct Buffers, and
+ * extract references to TypedArrays from direct Buffers.
+ */
+public final class TypedArrayBuffer {
+  private TypedArrayBuffer() {}
+
+  /**
+   * Wraps a JavaScript {@code ArrayBuffer} as a direct
+   * {@link java.nio.ByteBuffer}.
+   *
+   * <p>The provided {@code array} parameter must be a valid JavaScript
+   * {@code ArrayBuffer}, otherwise the behavior of this method is not
+   * specified.
+   *
+   * <p>The returned {@link java.nio.ByteBuffer} has the following properties:
+   *
+   * <ul>
+   *   <li>It has a {@code capacity()} equal to the {@code array.byteLength}.</li>
+   *   <li>Its initial {@code position()} is 0 and its {@code limit()} is its capacity.</li>
+   *   <li>It is a direct buffer backed by the provided {@code ArrayBuffer}:
+   *     changes to one are reflected on the other.</li>
+   * </ul>
+   *
+   * <p>{@code wrapArrayBuffer(array)} has the same behavior as
+   * {@code wrapInt8Array(new Int8Array(array))}, where {@code Int8Array} is
+   * the JavaScript class with that name.
+   *
+   * @param array a JavaScript {@code ArrayBuffer}
+   */
+  public static final ByteBuffer wrapArrayBuffer(Object array) {
+    throw new AssertionError("stub");
+  }
+
+  /**
+   * Wraps a view of a JavaScript {@code ArrayBuffer} as a direct
+   * {@link java.nio.ByteBuffer}.
+   *
+   * <p>The provided {@code array} parameter must be a valid JavaScript
+   * {@code ArrayBuffer}, otherwise the behavior of this method is not
+   * specified.
```

Comment:
Also clarify that the ArrayBuffer must have enough space?

Reply:
If it doesn't have enough space, a JavaScript `RangeError` exception is thrown, because `new Int8Array(array, byteOffset, length)` throws a `RangeError`. Do want me to make that explicit?

Reply:
Alternatively, we could move "up" the wrapping of the `ArrayBuffer` into an `Int8Array` within the scalajs-library. That would reduce the surface area of the javalib-intf.

Reply:
Yes, I think we should do that.

I have also been wondering whether we can remove some of the accessor methods. But I realized I do not sufficiently understand how the whole byte order business works :-/